include/quicly/sentmap.h

```diff
@@ -109,6 +114,12 @@ struct st_quicly_sent_t {
             int is_inflight;
             uint64_t generation;
         } new_token;
+        struct {
+            struct st_quicly_pending_new_cid_t *new_cid;
+        } new_connection_id;
+        struct {
+            struct st_quicly_pending_retire_cid_t *retire_cid;
```

Comment:
I totally like the idea to reduce (or cap) the amount of resource to track state. Just one question here:
>  If the list is full, we simply drop the CID without ever sending an RCID frame for it.

What will eventually happen to this CID? Do we ever get a chance to re-send it by being triggered by some other event? Peer including the sequence in retire-prior-to of NCID is one example, but what happens if retire-prior-to stays lower than the sequence number in question?

Reply:
> What will eventually happen to this CID?

From the viewpoint of the peer, it would look like it has not been retired. And I think that is fine. It would be the consequence of consuming CIDs too fast.

If that happens repeatedly on a single connection, the connection would eventually die. But it would be unlikely, as an endpoint should not be expecting the peer to issue CIDs at a high rate anyways; see https://quicwg.org/base-drafts/draft-ietf-quic-transport.html#section-5.1.1-5.

Reply:
Thanks for clarification. I'll go with your idea here then.

Reply:
Done.

Reply:
In fact I totally missed `quicly_get_first_timeout` in the initial version of the PR. Now it has been added. Thanks for pointing it out!